community: add Valthera integration

vselvarajijay opened the pull request:
```markdown
**Description:**  
This PR integrates Valthera into LangChain, introducing an framework designed to send highly personalized nudges by an LLM agent. This is modeled after Dr. BJ Fogg's Behavior Model. This integration includes:

- Custom data connectors for HubSpot, PostHog, and Snowflake.
- A unified data aggregator that consolidates user data.
- Scoring configurations to compute motivation and ability scores.
- A reasoning engine that determines the appropriate user action.
- A trigger generator to create personalized messages for user engagement.

**Issue:**  
N/A

**Dependencies:**  
N/A

**Twitter handle:**  
- `@vselvarajijay`

**Tests and Docs:**  
- `docs/docs/integrations/tools/valthera` 
- `https://github.com/valthera/langchain-valthera/tree/main/tests`

```

vselvarajijay committed:
doc updates

vselvarajijay committed:
+ fixing notebook

vselvarajijay committed:
fixing lint errors

vselvarajijay committed:
fixing lint issues

vselvarajijay committed:
fixing lint issues

vselvarajijay committed:
fixing lint issues

vselvarajijay committed:
Merge branch 'master' into master

vselvarajijay committed:
Fixing tools landing page

vselvarajijay committed:
fixing doc updates

vselvarajijay committed:
fixing lint issues

vselvarajijay committed:
fixing up messaging

vselvarajijay committed:
fixing copy

vselvarajijay commented:
Hey @ccurme - I would really appreciate a review on my first PR to LangChain. Happy to make any refinements if I missed anything. Appreciate the review and feedback! 🙂

ccurme commented on docs/docs/integrations/providers/valthera.mdx:
````diff
@@ -0,0 +1,25 @@
+1. **Install Valthera:**  
+   Install the package via pip:
+   ```bash
+   pip install valthera
````
We should document `langchain-valthera` in this page.

ccurme commented on docs/docs/integrations/providers/valthera.mdx:
````diff
@@ -0,0 +1,25 @@
+## Toolkits
+
+```python
+from valthera.tool import ValtheraTool
+from valthera.aggregator import DataAggregator
````
We should show imports from `langchain-valthera`.

ccurme reviewed (commented):
Thanks for this. I left some in-line comments on the provider page. Remaining things to do:
- Add `langchain-valthera` to the [package registry](https://github.com/langchain-ai/langchain/blob/master/libs/packages.yml).
- Revert changes to the integration template in `langchain-cli` and `uv.lock`.
- [Pypi page](https://pypi.org/project/langchain-valthera/) has some issues:
  - It documents `ChatValthera`, `ValtheraEmbeddings`, and other abstractions that [do not appear to exist](https://github.com/valthera/valthera/blob/main/langchain-valthera/langchain_valthera/__init__.py).
  - The homepage is the langchain monorepo instead of https://github.com/valthera/valthera.

vselvarajijay committed:
Adding Valthera to package registry

vselvarajijay committed:
Merge branch 'langchain-ai:master' into master

vselvarajijay committed:
Revert changes to tools.ipynb from commit e0b31f5

vselvarajijay committed:
Revert changes to uv.lock from commit f4b0975

vselvarajijay committed:
updating the provider doc

vselvarajijay committed:
fixing doc title

vselvarajijay commented:
@ccurme Thanks for the review! It looks like I had some old templates there, I cleared out the abstractions I wasn’t implementing.

1. `langchain-valthera` is now added to the package registry
2. Reverted template in `langchain-cli` and `uv.lock`
3. Fixed `Pypi` page:
    1. Removed references to unimplemented abstractions
    2. Fixed Homepage link
    3. Fixed Release Notes link
    4. Fixed Repository link
    5. Fixed Source code link
4. Corrected [docs/docs/integrations/providers/valthera.mdx](https://github.com/langchain-ai/langchain/pull/30105/files/e400b4902bcf3042bcdd097b95cfc1f5a97c41cd#diff-3455171b7e9c291f190b3dbcf48e20c87e01cb677a92c79ea62943afa2668f4a)

vselvarajijay committed:
Revert changes to uv.lock from commit 64d8580

vselvarajijay committed:
Rolling back uv.lock changes

vselvarajijay committed:
Merge branch 'master' into master

ccurme commented on docs/docs/integrations/tools/valthera.ipynb:
```diff
@@ -0,0 +1,439 @@
+   "source": [
+    "# Valthera\n",
+    "\n",
+    "Enable AI agents to engage users when they're most likely to.\n",
```
(nit) most likely to what?

vselvarajijay replied:
Hmm yeah good call. Hopefully this is better:

> Enable AI agents to engage users when they're most likely to respond.

vselvarajijay committed:
Keeping only LangChain example

vselvarajijay committed:
improving the doc title description

vselvarajijay commented on docs/docs/integrations/providers/valthera.mdx:
````diff
@@ -0,0 +1,154 @@
+print("✅ ValtheraTool successfully initialized for LangChain integration!")
+```
+
+### Example 2: Using ValtheraTool with LangChain Agents
````
Ok - cleared out the LangGraph examples and only keeping LangChain

vselvarajijay committed:
cleaning up some comments in the examples

vselvarajijay committed:
cleaning up example section structure

vselvarajijay committed:
fixing linting issue

vselvarajijay commented:
Hey @ccurme - a few more fixes on this PR as you suggested also a couple others I wanted to improve:

- Cleaned up integration tools doc
- Cleaned up provider doc
- Fixed up project structure of langchain-valthera (moved package to root of repo)
- Published new langchain-valthera package with updated links

Let me know if you think this is can be a good starting place for the Valthera tool. There is quite a bit more I would like to iterate on with the developer experience and capabilities that can help LLM Agents engage with people.

vselvarajijay committed:
Merge branch 'master' into master

ccurme committed:
Merge branch 'master' into vselvarajijay/master

# Conflicts:
#	libs/packages.yml

ccurme committed:
fix merge
